packages/bot-web-ui/src/stores/toolbox-store.ts

```diff
@@ -107,24 +107,25 @@ export default class ToolboxStore {
             this.is_workspace_scroll_adjusted = true;
 
             setTimeout(() => {
+                const workspace = window.Blockly.derivWorkspace;
                 const toolbox_width = document.getElementById('gtm-toolbox')?.getBoundingClientRect().width || 0;
-                const toolbar_width = document.querySelector('.toolbar__group-btn')?.getBoundingClientRect().width || 0;
                 const block_canvas_rect = workspace.svgBlockCanvas_?.getBoundingClientRect(); // eslint-disable-line
 
-                // Need to set the direction right if the workspace is RTL
-                const is_workspace_LTR = !workspace.RTL;
-                let canvas_direction = block_canvas_rect.left;
-                if (!is_workspace_LTR && window.innerWidth < 768) canvas_direction = block_canvas_rect.right;
+                if (workspace.RTL && block_canvas_rect) {
+                    let block_canvas_left = block_canvas_rect?.left;
```

Comment:
Dont think you be needing the optional chaining here since you are already checking for block_canvas_rect is not null or undefined in the if statement above. same for the bottom as well

Reply:
done ✅